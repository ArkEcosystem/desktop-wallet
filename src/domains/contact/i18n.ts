export const translations: { [key: string]: any } = {
	CONTACTS_PAGE: {
		ADD_CONTACT: "Add Contact",
		ADD_CONTACT_MESSAGE: "Add your most frequent contacts for fast, easy payments",
		TITLE: "My Contacts",
		SUBTITLE: "Manage your frequent contacts",
	},
	CONTACT_FORM: {
		ADD_ADDRESS: "Add address",
		ADDRESS: "Address",
		ADDRESSES: "Addresses",
		DELETE_CONTACT: "Delete Contact",
		NAME: "Name",
		NETWORK: "Network",
	},
	MODAL_CREATE_CONTACT: {
		TITLE: "Add Contact",
		DESCRIPTION: "Enter your contacts name and addresses and add them to your list.",
	},
	MODAL_DELETE_CONTACT: {
		TITLE: "Delete Contact",
		DESCRIPTION: "Do you really want to delete this contact? Once deleted, you will not be able to restore it.",
	},
	MODAL_UPDATE_CONTACT: {
		TITLE: "Edit Contact",
	},
	MODAL_SEARCH_CONTACT: {
		TITLE: "Search Contact",
		DESCRIPTION: "Find the contact you need quickly and conveniently",
		PLACEHOLDER: "Search...",
	},
	VALIDATION: {
		MAXLENGTH_ERROR: "The Contact Name should have less than {{maxLength}} characters.",
		CONTACT_NAME_EXISTS: "Contact with name {{name}} already exists.",
		CONTACT_NAME_EXISTS_SUFFIX: "already exists",
	},
};
