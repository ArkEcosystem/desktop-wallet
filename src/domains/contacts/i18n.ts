export const translations: { [key: string]: any } = {
	CONTACT_FORM: {
		ADD_ADDRESS: "Add Address",
		ADDRESS: "Address",
		ADDRESSES: "Addresses",
		NAME: "Name",
		NETWORK: "Network",
	},
	MODAL_CREATE_CONTACT: {
		TITLE: "Add Contact",
		DESCRIPTION: "Enter your contacts name and addresses and add them to your list.",
	},
	MODAL_DELETE_CONTACT: {
		TITLE: "Delete Contact",
		DESCRIPTION: "Do you really want to delete this contact? Once deleted, you will not be able to restore it.",
	},
	MODAL_UPDATE_CONTACT: {
		TITLE: "Edit Contact",
	},
};
