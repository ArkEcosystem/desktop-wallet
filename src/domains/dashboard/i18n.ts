export const translations: { [key: string]: any } = {
	FILTER_WALLETS: {
		NETWORK: {
			TITLE: "Filter Network",
			DESCRIPTION: "Select the types of networks",
		},
		WALLETS: {
			TITLE: "Your Wallets",
			DESCRIPTION: "Select the display of your wallets",
		},
		CHART: {
			TITLE: "Portfolio Chart",
			DESCRIPTION: "You can hide this unit if you like",
		},
		TRANSACTION_HISTORY: {
			TITLE: "Transaction History",
			DESCRIPTION: "You can hide this unit if you like",
		},
	},

	TRANSACTION_HISTORY: {
		TITLE: "Transactions History",
		EMPTY_TEXT:
			"This will display the history of your transactions. But you don't have more than one transaction at the moment.",
	},

	WALLET_CONTROLS: {
		CREATE: "Create",
		IMPORT: "Import",
	},

	DASHBOARD_PAGE: {
		CHART: {
			TITLE: "Portfolio Chart",
			PERCENTAGES_LABEL: "Total portfolio",
		},
	},
};
