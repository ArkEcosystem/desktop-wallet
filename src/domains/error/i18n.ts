export const translations: { [key: string]: any } = {
	APPLICATION: {
		TITLE: "Something went wrong",
		DESCRIPTION: "It looks like something went wrong. Try visiting another page or reloading.",
		RELOAD: "Reload",
	},
	OFFLINE: {
		TITLE: "Cannot Connect",
		DESCRIPTION: "Check your internet connection if it doesn't work, check it again.",
	},
};
