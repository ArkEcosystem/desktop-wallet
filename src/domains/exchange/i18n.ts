export const translations: { [key: string]: any } = {
	MODAL_ADD_EXCHANGE: {
		TITLE: "Exchange",
		DESCRIPTION:
			"Buy ARK directly from the ARK Desktop Wallet or through one of 20+ exchanges supplying ARK tokens using BTC, ETH or other trading pairs.",
	},
};
