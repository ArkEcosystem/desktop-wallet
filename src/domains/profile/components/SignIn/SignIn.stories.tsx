import { ARK } from "@arkecosystem/platform-sdk-ark";
import { Environment } from "@arkecosystem/platform-sdk-profiles";
import { action } from "@storybook/addon-actions";
import { boolean, withKnobs } from "@storybook/addon-knobs";
import { EnvironmentProvider } from "app/contexts";
import { httpClient } from "app/services";
import React from "react";
import { StubStorage } from "tests/mocks";

import { SignIn } from "./SignIn";

export default {
	title: "Domains / Profile / Components / SignIn",
	decorators: [withKnobs],
};

export const Default = () => {
	const env = new Environment({ coins: { ARK }, httpClient, storage: new StubStorage() });
	const profile = env.profiles().create("Test profile");
	profile.auth().setPassword("password");

	return (
		<EnvironmentProvider env={env}>
			<SignIn
				isOpen={boolean("isOpen", true)}
				profile={profile}
				onCancel={action("onCancel")}
				onClose={action("onClose")}
				onSuccess={action("onSuccess")}
			/>
		</EnvironmentProvider>
	);
};
