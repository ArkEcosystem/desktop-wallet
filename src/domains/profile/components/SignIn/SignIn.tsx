import { MemoryPassword, Profile } from "@arkecosystem/platform-sdk-profiles";
import { Button } from "app/components/Button";
import { Divider } from "app/components/Divider";
import { Form, FormField, FormLabel } from "app/components/Form";
import { InputPassword } from "app/components/Input";
import { Modal } from "app/components/Modal";
import { ProfileAvatar } from "domains/profile/components/ProfileAvatar";
import React, { useCallback, useEffect, useState } from "react";
import { useForm } from "react-hook-form";
import { useTranslation } from "react-i18next";

type SignInProps = {
	isOpen: boolean;
	profile: Profile;
	onCancel?: any;
	onClose?: any;
	onSuccess: (password: string) => void;
};

const MAX_ATTEMPTS = 3;
const TIMEOUT = 60;

export const SignIn = ({ isOpen, profile, onCancel, onClose, onSuccess }: SignInProps) => {
	const { t } = useTranslation();

	const methods = useForm({ mode: "onChange" });
	const { errors, formState, register, setError } = methods;

	const [count, setCount] = useState(0);
	const [remainingTime, setRemainingTime] = useState(0);

	const hasReachedLimit = useCallback(() => count >= MAX_ATTEMPTS, [count]);

	useEffect(() => {
		if (hasReachedLimit()) {
			setRemainingTime(TIMEOUT);
		}

		if (count) {
			const timer = setInterval(() => {
				setCount(count - 1);
			}, TIMEOUT * 1000);

			return () => clearInterval(timer);
		}
	}, [count, hasReachedLimit]);

	useEffect(() => {
		if (remainingTime) {
			setError("password", {
				type: "maxAttempts",
				message: t("PROFILE.MODAL_SIGN_IN.MAX_ATTEMPTS_ERROR", { remainingTime }),
			});

			const timer = setInterval(() => {
				setRemainingTime(remainingTime - 1);
			}, 1000);

			return () => clearInterval(timer);
		} else if (errors.password) {
			setError("password", {
				type: "invalid",
				message: t("COMMON.VALIDATION.SUBJECT_INVALID", {
					subject: t("COMMON.PASSWORD"),
				}),
			});
		}
	}, [errors, remainingTime, setError, t]);

	const handleSubmit = ({ password }: any) => {
		if (profile.auth().verifyPassword(password)) {
			MemoryPassword.set(profile, password);
			onSuccess(password);
		} else {
			setCount(count + 1);

			setError("password", {
				type: "invalid",
				message: t("COMMON.VALIDATION.SUBJECT_INVALID", {
					subject: t("COMMON.PASSWORD"),
				}),
			});
		}
	};

	return (
		<Modal
			title={t("PROFILE.MODAL_SIGN_IN.TITLE")}
			description={t("PROFILE.MODAL_SIGN_IN.DESCRIPTION")}
			size="md"
			isOpen={isOpen}
			onClose={onClose}
		>
			<div className="mt-8">
				<div className="flex items-center space-x-3">
					<ProfileAvatar profile={profile} />

					<div className="flex flex-col">
						<p className="text-sm font-semibold text-theme-secondary-700">{t("COMMON.NAME")}</p>
						<p className="font-semibold text-theme-text" data-testid="profile-card__user--name">
							<span>{profile.name()}</span>
						</p>
					</div>
				</div>
			</div>

			<Divider dashed={true} />

			<Form context={methods} onSubmit={handleSubmit} className="mt-8">
				<FormField name="password">
					<FormLabel label={t("SETTINGS.GENERAL.PERSONAL.PASSWORD")} required />
					<InputPassword
						ref={register({
							required: t("COMMON.VALIDATION.FIELD_REQUIRED", {
								field: t("COMMON.PASSWORD"),
							}).toString(),
						})}
						disabled={hasReachedLimit()}
						data-testid="SignIn__input--password"
					/>
				</FormField>

				<div className="flex justify-end mt-8 space-x-3">
					<Button data-testid="SignIn__cancel-button" variant="secondary" onClick={onCancel}>
						{t("COMMON.CANCEL")}
					</Button>

					<Button data-testid="SignIn__submit-button" type="submit" disabled={!formState.isValid}>
						{t("COMMON.SIGN_IN")}
					</Button>
				</div>
			</Form>
		</Modal>
	);
};
