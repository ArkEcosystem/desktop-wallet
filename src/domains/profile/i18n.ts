export const translations: { [key: string]: any } = {
	MODAL_PROFILE_CREATED: {
		TITLE: "Profile Created!",
		DESCRIPTION_1: "If you are new to the ARK Desktop Wallet, view our tutorial to get started.",
		DESCRIPTION_2: "If not, you can skip the tutorial to go directly to your portfolio.",
		START_TUTORIAL: "Start Tutorial",
		SKIP_TUTORIAL: "Skip Tutorial",
	},
};
