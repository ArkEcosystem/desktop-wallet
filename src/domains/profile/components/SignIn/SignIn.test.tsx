/* eslint-disable @typescript-eslint/require-await */
import { Profile } from "@arkecosystem/platform-sdk-profiles";
import React from "react";
import { act, env, fireEvent, renderWithRouter, waitFor } from "testing-library";

import { translations } from "../../i18n";
import { SignIn } from "./SignIn";

let profile: Profile;

jest.setTimeout(30000);

describe("SignIn", () => {
	beforeEach(async () => {
		profile = env.profiles().findById("cba050f1-880f-45f0-9af9-cfe48f406052");
	});

	beforeEach(() => {
		jest.useFakeTimers();
	});

	afterAll(() => {
		jest.useRealTimers();
	});

	it("should not render if not open", () => {
		const { asFragment, getByTestId } = renderWithRouter(<SignIn profile={profile} isOpen={false} />);

		expect(() => getByTestId("modal__inner")).toThrow(/Unable to find an element by/);
		expect(asFragment()).toMatchSnapshot();
	});

	it("should render a modal", async () => {
		let renderContext: any;

		await act(async () => {
			renderContext = renderWithRouter(<SignIn isOpen={true} profile={profile} />);
		});

		const { asFragment, getByTestId } = renderContext;

		expect(getByTestId("modal__inner")).toHaveTextContent(translations.MODAL_SIGN_IN.TITLE);
		expect(getByTestId("modal__inner")).toHaveTextContent(translations.MODAL_SIGN_IN.DESCRIPTION);
		expect(asFragment()).toMatchSnapshot();
	});

	it("should cancel sign in", async () => {
		const onCancel = jest.fn();

		const { getByTestId } = renderWithRouter(<SignIn isOpen={true} profile={profile} onCancel={onCancel} />);

		act(() => {
			fireEvent.click(getByTestId("SignIn__cancel-button"));
		});

		await waitFor(() => {
			expect(onCancel).toBeCalled();
		});
	});
	it("should call onSuccess callback", async () => {
		const onSuccess = jest.fn();

		const { findByTestId, getByTestId } = renderWithRouter(
			<SignIn isOpen={true} profile={profile} onSuccess={onSuccess} />,
		);

		await act(async () => {
			fireEvent.input(getByTestId("SignIn__input--password"), { target: { value: "password" } });
		});

		// wait for formState.isValid to be updated
		await findByTestId("SignIn__submit-button");

		act(() => {
			fireEvent.click(getByTestId("SignIn__submit-button"));
		});

		await waitFor(() => {
			expect(onSuccess).toBeCalled();
		});
	});

	it("should set an error if the password is invalid", async () => {
		const onSuccess = jest.fn();

		const { findByTestId, getByTestId, queryByText } = renderWithRouter(
			<SignIn isOpen={true} profile={profile} onSuccess={onSuccess} />,
		);

		await act(async () => {
			fireEvent.input(getByTestId("SignIn__input--password"), { target: { value: "wrong password" } });
		});

		// wait for formState.isValid to be updated
		await findByTestId("SignIn__submit-button");

		act(() => {
			fireEvent.click(getByTestId("SignIn__submit-button"));
		});

		// wait for formState.isValid to be updated
		await findByTestId("SignIn__submit-button");

		expect(getByTestId("Input-error")).toBeVisible();
		expect(getByTestId("SignIn__submit-button")).toBeDisabled();
	});

	it("should set an error and disable the input if the password is invalid multiple times", async () => {
		const onSuccess = jest.fn();

		let renderContext: any;

		await act(async () => {
			renderContext = renderWithRouter(<SignIn isOpen={true} profile={profile} onSuccess={onSuccess} />);
		});

		const { findByTestId, getByTestId, queryByText } = renderContext;

		for (const i of [1, 2, 3]) {
			await act(async () => {
				fireEvent.input(getByTestId("SignIn__input--password"), { target: { value: `wrong password ${i}` } });
			});

			// wait for form to be updated
			await findByTestId("SignIn__submit-button");

			act(() => {
				fireEvent.click(getByTestId("SignIn__submit-button"));
			});

			// wait for form to be updated
			await findByTestId("SignIn__submit-button");
		}

		expect(getByTestId("SignIn__submit-button")).toBeDisabled();
		expect(getByTestId("SignIn__input--password")).toBeDisabled();

		act(() => {
			jest.advanceTimersByTime(65000);
			jest.clearAllTimers();
		});

		// wait for form to be updated
		await findByTestId("SignIn__submit-button");

		await waitFor(
			() => expect(getByTestId("Input-error")).toHaveAttribute("data-errortext", "The Password is invalid"),
			{ timeout: 10000 },
		);
	});
});
