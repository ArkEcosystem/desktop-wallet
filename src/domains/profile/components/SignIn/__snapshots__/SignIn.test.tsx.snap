// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn should not render if not open 1`] = `<DocumentFragment />`;

exports[`SignIn should render a modal 1`] = `
<DocumentFragment>
  <div
    class="flex fixed inset-0 z-50 justify-center items-center w-full h-full overflow-overlay"
  >
    <div
      class="fixed z-50 w-full h-full bg-black opacity-50"
      data-testid="modal__overlay"
    />
    <div
      class="Modal__ModalContainer-sc-1pv7yg7-0 kPeakM absolute left-0 right-0 z-50 flex flex-col p-10 mx-auto overflow-hidden rounded-2xl bg-theme-background shadow-2xl "
      data-testid="modal__inner"
    >
      <div
        class="absolute top-0 right-0 z-50 mt-5 mr-5 rounded transition-all duration-100 ease-linear bg-theme-primary-100 hover:bg-theme-primary-300 dark:bg-theme-secondary-800 dark:text-theme-secondary-600 dark:hover:bg-theme-secondary-700 dark:hover:text-theme-secondary-400"
      >
        <button
          class="Button__StyledButton-mdtvtf-0 fHYGyx w-11 h-11"
          data-testid="modal__close-btn"
          type="button"
        >
          <div
            class="flex relative items-center space-x-2"
          >
            <div
              class="sc-bdfBwQ bfBhNR"
              height="14"
              width="14"
            >
              <svg>
                cross-slim.svg
              </svg>
            </div>
          </div>
        </button>
      </div>
      <div
        class="relative"
      >
        <h2
          class="mb-0 text-3xl font-bold"
        >
          Sign In
        </h2>
        <div
          class="flex-1"
        >
          <div
            class="mt-1 text-theme-secondary-text whitespace-pre-line"
          >
            Sign in to access your profile.
          </div>
          <div
            class="mt-8"
          >
            <div
              class="flex items-center space-x-3"
            >
              <div
                class="Avatar__AvatarWrapper-sc-1vsy2s2-0 gyYbkb"
                data-testid="ProfileAvatar__svg"
              >
                <img
                  alt="Jane Doe"
                  src="data:image/svg+xml;utf8,<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"picasso\\" width=\\"100\\" height=\\"100\\" viewBox=\\"0 0 100 100\\"><style>.picasso circle{mix-blend-mode:soft-light;}</style><rect fill=\\"rgb(0, 150, 136)\\" width=\\"100\\" height=\\"100\\"/><circle r=\\"55\\" cx=\\"20\\" cy=\\"60\\" fill=\\"rgb(233, 30, 99)\\"/><circle r=\\"60\\" cx=\\"50\\" cy=\\"70\\" fill=\\"rgb(255, 193, 7)\\"/><circle r=\\"45\\" cx=\\"30\\" cy=\\"30\\" fill=\\"rgb(63, 81, 181)\\"/></svg>"
                  title="Jane Doe"
                />
                <span
                  class="absolute font-semibold text-white"
                >
                  JA
                </span>
              </div>
              <div
                class="flex flex-col"
              >
                <p
                  class="text-sm font-semibold text-theme-secondary-700"
                >
                  Name
                </p>
                <p
                  class="font-semibold text-theme-text"
                  data-testid="profile-card__user--name"
                >
                  <span>
                    Jane Doe
                  </span>
                </p>
              </div>
            </div>
          </div>
          <div
            class="Divider-sc-19q4mlo-0 bKsVMa border-theme-secondary-300 dark:border-theme-secondary-800"
            type="horizontal"
          />
          <form
            class="mt-8"
            data-testid="Form"
          >
            <fieldset
              class="FormField__FormFieldStyled-yeauwd-0 cnlnjc flex flex-col space-y-2"
            >
              <label
                class="inline-block flex mb-2 text-sm font-semibold transition-colors duration-100 FormLabel text-theme-secondary-700"
                data-testid="FormLabel"
                for="password"
              >
                Password
                <div
                  class="mt-1 ml-1 w-1 h-1 rounded-full bg-theme-danger-400"
                  data-testid="FormLabel__required"
                />
              </label>
              <div
                class="InputGroup__InputGroupStyle-sc-1obbdqe-3 iqafwD max-w-20"
                data-testid="InputPassword"
              >
                <input
                  class="Input__InputStyled-wu99qs-0 eVdxgh overflow-hidden w-full bg-theme-background appearance-none rounded border border-theme-secondary-400 dark:border-theme-secondary-700 text-theme-text transition-colors duration-200 px-4 py-3 no-ligatures pr-12"
                  data-testid="SignIn__input--password"
                  name="password"
                  type="password"
                />
                <div
                  class="InputGroup__InputAddonStyle-sc-1obbdqe-0 InputGroup__InputAddonEndStyle-sc-1obbdqe-2 epSJiV bWrKSO my-px mr-4"
                >
                  <button
                    class="flex justify-center items-center w-full h-full text-2xl focus:outline-none text-theme-primary-300 dark:text-theme-secondary-600"
                    data-testid="InputPassword__toggle"
                    type="button"
                  >
                    <div
                      class="sc-bdfBwQ fPrsah"
                      height="1em"
                      width="1em"
                    >
                      <svg>
                        eye.svg
                      </svg>
                    </div>
                  </button>
                </div>
              </div>
            </fieldset>
            <div
              class="flex justify-end mt-8 space-x-3"
            >
              <button
                class="Button__StyledButton-mdtvtf-0 iUSBMV"
                data-testid="SignIn__cancel-button"
                type="button"
              >
                <div
                  class="flex relative items-center space-x-2"
                >
                  Cancel
                </div>
              </button>
              <button
                class="Button__StyledButton-mdtvtf-0 kwcZQn"
                data-testid="SignIn__submit-button"
                disabled=""
                type="submit"
              >
                <div
                  class="flex relative items-center space-x-2"
                >
                  Sign In
                </div>
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
